app/src/test/java/com/edulectronics/tinycircuit/CircuitTest/CircuitTest.java

```diff
@@ -36,4 +43,58 @@ public void removeComponent(){
 
         assertEquals(false, c.circuit.occupied(5));
     }
+
+    @Test
+    public void parallelCircuitNoResistance() {
+        ArrayList<Component> components = new ArrayList<Component>();
+
+        Powersource powersource = new Powersource();
+        Resistor resistor1 = new Resistor();
+        Lightbulb bulb = new Lightbulb();
+
+        // Connect lightbulb directly to powersource
+        Connector.connect(powersource.getInput(), bulb.getConnectionPointByIndex(0));
+        Connector.connect(powersource.getOutput(), bulb.getConnectionPointByIndex(1));
+
+        // And the resistor also, in parallel, directly to output.
+        Connector.connect(powersource.getInput(), resistor1.getConnectionPointByIndex(0));
+        Connector.connect(powersource.getOutput(), resistor1.getConnectionPointByIndex(1));
+
+        components.add(resistor1);
+        components.add(powersource);
+        components.add(bulb);
+        CircuitController controller = CircuitBuilder.getCircuitController(components);
+
+        controller.run();
+        assertTrue(bulb.isBroken());
+    }
+
+    @Test
+    public void parallelCircuitWithSomeResistance() {
+        ArrayList<Component> components = new ArrayList<Component>();
```

Comment:
Alle onderstaande code gebruik je twee keer. Misschien is het beter om een @Before methode te maken. De documentatie vind je hier: [http://junit.sourceforge.net/javadoc/org/junit/Before.html](http://junit.sourceforge.net/javadoc/org/junit/Before.html)

`ArrayList<Component> components = new ArrayList<Component>();`
`Powersource powersource = new Powersource();`
`Resistor resistor = new Resistor();`
`Lightbulb bulb = new Lightbulb();`
`Lightbulb bulb2 = new Lightbulb();`
`// Connect lightbulb directly to powersource`
`Connector.connect(powersource.getInput(), bulb.getConnectionPointByIndex(0));`
`Connector.connect(powersource.getOutput(), bulb.getConnectionPointByIndex(1));`
`components.add(resistor);`
`components.add(powersource);`
`components.add(bulb);`
`CircuitController controller = CircuitBuilder.getCircuitController(components);`
`controller.run();`